src/schemaGenerator.ts

```diff
@@ -469,6 +478,30 @@ function addResolveFunctionsToSchema(
   checkForResolveTypeResolver(schema, requireResolversForResolveType);
 }
 
+function extendResolversFromInterfaces(schema: GraphQLSchema, resolvers: IResolvers) {
+  const typeNames = new Set([
+    ...Object.keys(schema.getTypeMap()),
+    ...Object.keys(resolvers)
+  ]);
+
+  const extendedResolvers: IResolvers = {};
+  typeNames.forEach((typeName) => {
+    const typeResolvers = resolvers[typeName];
+    const type = schema.getType(typeName);
+    if (!(type instanceof GraphQLObjectType)) {
+      if (typeResolvers) {
+        extendedResolvers[typeName] = typeResolvers;
+      }
+      return;
+    }
+
+    const interfaceResolvers = (type as GraphQLObjectType).getInterfaces().map((iFace) => resolvers[iFace.name]);
```

Comment:
You might be able to reorganize this code so that the `type.getInterfaces()` call is inside a `type instanceof GraphQLObject` conditional block, so you don't have to explicitly cast it. Then you could put the code above in the `else` branch and avoid the early `return`, too.